components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -227,28 +227,30 @@ static void psa_mac_operation(void)
 
             switch (psa_crypto.func) {
                 case PSA_MAC_SIGN_SETUP: {
-                    if (!psa_crypto_access_control_is_handle_permitted(psa_crypto.handle,
-                                                                       msg.client_id)) {
+                    if (psa_crypto_access_control_is_handle_permitted(psa_crypto.handle, msg.client_id)) {
+                        status = psa_mac_sign_setup(msg.rhandle, psa_crypto.handle, psa_crypto.alg);
+                    } else {
                         status = PSA_ERROR_INVALID_HANDLE;
-                        break;
                     }
 
-                    status = psa_mac_sign_setup(msg.rhandle,
-                                                psa_crypto.handle,
-                                                psa_crypto.alg);
+                    if (status != PSA_SUCCESS) {
+                        mbedtls_free(msg.rhandle);
```

Comment:
is it ok to free handle (mbedtls_free) for both scenarios:
sign setup fails
handle is not permitted?